src/main/java/io/javalin/core/JavalinServer.kt

```diff
@@ -60,41 +58,41 @@ class JavalinServer {
                 jettyRequest.isHandled = true
             }
         }.apply {
-            this.sessionHandler = jettySessionHandler
+            this.sessionHandler = config.sessionHandler
         }
 
-        val webSocketHandler = ServletContextHandler(nullParent, contextPath).apply {
+        val webSocketHandler = ServletContextHandler(nullParent, config.contextPath).apply {
             addServlet(ServletHolder(javalinWsServlet), "/*")
         }
 
         val notFoundHandler = object : SessionHandler() {
             override fun doHandle(target: String, jettyRequest: Request, request: HttpServletRequest, response: HttpServletResponse) {
-                val msg = "Not found. Request is below context-path (context-path: '$contextPath')"
+                val msg = "Not found. Request is below context-path (context-path: '${config.contextPath}')"
                 response.status = 404
                 ByteArrayInputStream(msg.toByteArray()).copyTo(response.outputStream)
                 response.outputStream.close()
-                Javalin.log.warn("Received a request below context-path (context-path: '$contextPath'). Returned 404.")
+                Javalin.log.warn("Received a request below context-path (context-path: '${config.contextPath}'). Returned 404.")
             }
         }
 
-        jettyServer.apply {
+        config.server!!.apply {
             handler = attachJavalinHandlers(server.handler, HandlerList(httpHandler, webSocketHandler, notFoundHandler))
             connectors = connectors.takeIf { it.isNotEmpty() } ?: arrayOf(ServerConnector(server).apply {
-                this.port = jettyPort
+                this.port = config.port
             })
         }.start()
 
-        jettyServer.connectors.filterIsInstance<ServerConnector>().forEach {
-            Javalin.log.info("Listening on ${it.protocol}://${it.host ?: "localhost"}:${it.localPort}$contextPath")
+        config.server!!.connectors.filterIsInstance<ServerConnector>().forEach {
+            Javalin.log.info("Listening on ${it.protocol}://${it.host ?: "localhost"}:${it.localPort}${config.contextPath}")
         }
 
-        jettyServer.connectors.filter { it !is ServerConnector }.forEach {
+        config.server!!.connectors.filter { it !is ServerConnector }.forEach {
```

Comment:
`config.server!!` can be replaced with `config.server` (there are more occurrences of this) since the field is not nullable.

Reply:
Thanks, fixed!